api_volontaria/apps/position/models.py

```diff
@@ -0,0 +1,184 @@
+
+    @staticmethod
+    def has_create_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_destroy_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_update_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @authenticated_users
+    def has_object_destroy_permission(self, request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @authenticated_users
+    def has_object_update_permission(self, request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+
+class Application(models.Model):
+    """
+    This class represents an application made by volunteer for a given position.
+    """
+
+    APPLICATION_UNDER_EXAMINATION = 'UNDER_EXAMINATION'
+    APPLICATION_ACCEPTED = 'ACCEPTED'
+    APPLICATION_DECLINED = 'DECLINED'
+
+    APPLICATION_CHOICES = (
+        (APPLICATION_UNDER_EXAMINATION, _('Under examination')),
+        (APPLICATION_ACCEPTED, _('Accepted')),
+        (APPLICATION_DECLINED, _('Declined')),
+    )
+
+    class Meta:
+        verbose_name = _("Application")
+        verbose_name_plural = _("Applications")
+
+    position = models.ForeignKey(
+        Position,
+        verbose_name=_("Position"),
+        blank=False,
+        on_delete=models.CASCADE,
+    )
+    
+    applied_on = models.DateTimeField(
+        verbose_name=_("Applied on"),
+        auto_now_add=True,
+    )
+    
+    motivation = models.TextField(
+        verbose_name="Motivation",
+    )
+
+    application_status = models.CharField(
+        verbose_name=_("Application status"),
+        max_length=100,
+        choices=APPLICATION_CHOICES,
+        default=APPLICATION_UNDER_EXAMINATION
+    )
+
+    user = models.ForeignKey(
+        User,
+        related_name='applications',
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return self.application.name
+        # return self.position.name
+
+    @staticmethod
+    def has_destroy_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_update_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_list_permission(request):
+        return True
+
+    @staticmethod
+    def has_create_permission(request):
+        return True
+
+    @authenticated_users
+    def has_object_update_permission(self, request):
+        if self.user == request.user:
+            return True
+        if request.user.is_staff:
+            return True
+        else:
+            return False
```

Comment:
I think that user should be able to modify his own application ONLY IF the application is always under review. After that it become like a contract between the admin and the user and if the user can change it by himself it could be problematic.

What do you think ?